.gitignore

```diff
@@ -13,3 +13,6 @@ docs/build/
 
 # Hatch generated files
 dist/
+
+# venvs
+*venv/
```

Comment:
Is there a reason the venvs would be in the source directory? (I think I've always studiously avoided having venvs or builds in a source directory, to the extent I could.)

Reply:
TBH I agree; I keep my virtual envs in a centralized folder. However, a lot of IDEs will create the virtual env in the project folder by default. It's also much easier to give instructions for setting up an environment if assume the venv is created locally. Given that, it seems worth excluding the `*venv` pattern, thus supporting people to create venv within the project, unless there is a good reason to preserve it.

Another reason that isn't relevant in this case, is allows for self-contained `make` targets for the project, where that is popular (`make test`, `make clean`, `make build`, etc).